builders/build_capo.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+help() {
+    echo "Build a CAPO Image"
+    echo ""
+    echo "Usage: ./build_capo.sh [options] <quay.io username>"
+    echo "Options:"
+    echo "-h, --help      show this message"
+    echo "-t, --tag       push to a custom tag in your origin release image repo, default: capo"
+    echo ""
+}
+
+TAG="capo"
+
+# Parse Options
+case $1 in
+    -h|--help)
+        help
+        exit 0;;
+    -t|--tag)
+        TAG=$2
+        shift
+        shift;;
+    *);;
+esac
+
+if [ -z "$1" ]; then
+    echo "No quay.io username provided, exiting ..."
+    exit 1
+fi
+
+if [ ! -f config.json ]; then
+    echo "config.json not found, exiting ..."
```

Comment:
I find this config.json a bit surprising. It forces me to keep secrets alongside code, and I am pretty sure I will forget to update this file when the `registry.svc.ci.openshift.org` token expires.

Moreover, I already have an openshift-themed pull-secret ready, that I use for running `make update` on the `release` repo for example. What about accepting a pull secret with the env var `OC_REGISTRY_AUTH_FILE` (to mimic Podman's `REGISTRY_AUTH_FILE`), instead of creating `config.json`?

The ideal solution though would be to find a way around this double-pull-secret madness.

Reply:
yep, it makes sense, no you can provide a path to your auth file either with --auth option or by setting OC_REGISTRY_AUTH_FILE env variable. By default it will be config.json in the local folder